packages/unimodules-permissions-interface/ios/UMPermissionsInterface/UMPermissionsInterface.h

```diff
@@ -1,18 +1,43 @@
 // Copyright 2018-present 650 Industries. All rights reserved.
 
 #import <Foundation/Foundation.h>
+#import <UMCore/UMModuleRegistry.h>
+
+typedef enum UMPermissionStatus {
+  UMPermissionStatusDenied,
+  UMPermissionStatusGranted,
+  UMPermissionStatusUndetermined,
+} UMPermissionStatus;
+
+
+@protocol UMPermissionsRequester <NSObject>
+
++ (NSString *)permissionType;
+
+- (void)requestPermissionsWithResolver:(UMPromiseResolveBlock)resolve rejecter:(UMPromiseRejectBlock)reject;
+
+- (NSDictionary *)getPermissions;
+
+@end
 
 @protocol UMPermissionsInterface
 
 - (NSDictionary *)getPermissionsForResource:(NSString *)resource;
+
 - (BOOL)hasGrantedPermission:(NSString *)permissionType;
 
 - (void)askForPermission:(NSString *)permissionType
               withResult:(void (^)(NSDictionary *))onResult
```

Comment:
I wonder whether this actually cannot be
```suggestion
              withResult:(UMPromiseResolveBlock)onResult
```
❓